addons/mail/static/src/js/composer.js

```diff
@@ -469,7 +469,16 @@ var BasicComposer = Widget.extend(chat_mixin, {
     preprocess_message: function () {
         // Return a deferred as this function is extended with asynchronous
         // behavior for the chatter composer
-        var value = _.escape(this.$input.val()).replace(/\n|\r/g, '<br/>');
+
+        //Remove Extra newlines from starting and ending of Message.
+        var value = _.escape(this.$input.val()).replace(/^(\r|\n)+|(\n|\r)+$/g, '');
+
+        //Combine 2 or more consecutive newlines into 1 newline.
+        value = value.replace(/(\r|\n){2,}/g, '<br/><br/>');
+
+        //convert newline to <br/> tag.
+        value = value.replace(/(\r|\n){1,}/g, '<br/>');
```

Comment:
Isn't the + in 1+ already handled with the 2+ line above? Also comment is not really useful since it repeats exactly what the code does.